src/systems/filecoin_vm/runtime/impl/runtime.go

```diff
@@ -308,6 +309,38 @@ func (rt *VMContext) _checkNumValidateCalls(x int) {
 	}
 }
 
+func (rt *VMContext) ValidateIsStorageMiner(minerAddr addr.Address) bool {
+	codeID, ok := rt.GetActorCodeID(minerAddr)
+	Assert(ok)
+	if !codeID.IsBuiltin() {
+		return false
+	}
+	return (codeID.As_Builtin() == actor.BuiltinActorID_StorageMiner)
+}
```

Comment:
Strong objection! Actors should not inspect the code type of other actors. This is terribly brittle and a crutch for things which can be better represented by actors themselves.
This is also redundant given that #753 gives access to the actor code ID.